### PR TITLE
Ruby Blog: Validation and Spec

### PR DESCRIPTION
**Pull request by AmanWorku**
## Hey there....

# In this PR I:
- [x] Added validations for the User model
- [x] Added validations for the Post model with requirements given.
- [x] Added unit specs for all of the models' methods and validations.  

_Cheers🥂_

### CODE REVIEW

File: spec/model/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
```
Comment (Ismailco) [inline spec/model/user_spec.rb:3]: - [ ] Any custom methods in the `User` model should have a corresponding unit test, kindly make sure to implement tests for the `recent_three_posts` your Model is fully tested

File: spec/model/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    expect(user).to_not be_valid
+  end
+  it 'Post counter number must be numeric' do
+    user.posts_counter = ''
+    expect(user).to_not be_valid
+  end
+  it 'Post counter number must be greater than 0' do
+    user.posts_counter = -10
+    expect(user).to_not be_valid
+  end
```
Comment (Ismailco) [inline spec/model/user_spec.rb:16]: - [ ] [OPTIONAL] Kindly make sure to include more cases in your unit tests, like when the user is present, post_conter has a valid value (positive number)...

File: spec/model/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
```
Comment (Ismailco) [inline spec/model/post_spec.rb:3]: - [ ] Same here, any custom methods in the `Post` model should have a corresponding unit test, kindly make sure to implement tests for the `update_posts_count` your Model is fully tested

File: spec/model/like_spec.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
```
Comment (Ismailco) [inline spec/model/like_spec.rb:3]: - [ ] Kindly make sure to include tests for `update_like_count` method so your Model is fully tested

File: spec/model/comment_spec.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
```
Comment (Ismailco) [inline spec/model/comment_spec.rb:3]: - [ ] Kindly make sure to include tests for `update_comment_count` method so your Model is fully tested

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @AmanWorku, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- Great job implementing all the required tests changes requested by the previous reviewer :heavy_check_mark: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AmanWorku** [comment]
Thank you so much @CollinsTatang. Have a great day.

**Pull request merged**